exercise-app.js

```diff
@@ -28,13 +19,59 @@ const Weather = ({ temp, description, icon }) => (
 );
 
 
-const App = () => (
+class App extends React.Component {
+
+  constructor(props) {                // where do we hook up these props to ? where does this prop data come from?
+  super(props);                       // what is the purpose of this? why do we need to call super?
+  this.state = {
+  temp: 0,
+  description: 'not loaded yet',
+  icon: `http://openweathermap.org/img/w/10d.png`};  // always intialize a state? is the state to be updated by passing in props?
```

Comment:
It's a good idea to initialize state. Although some values might be set to `null` or `undefined`. It allows people to quickly figure out what all data is being managed as state. As for the initialization values… depends on the UX you are going for. If you want to show a loading spinner while the data is loading then probably use `null`. If you want to show a placeholder image then use that as initial state.